packages/ra-ui-materialui/src/button/UpdateWithUndoButton.spec.tsx

```diff
@@ -95,11 +96,23 @@ describe('<UpdateWithUndoButton />', () => {
                     id: 123,
                     data: { views: 0 },
                     meta: undefined,
-                    previousData: { id: 123, title: 'lorem', views: 1000 },
+                    previousData: { id: 123, title: 'lorem', views: 500 },
                     resource: 'posts',
                 },
                 { snapshot: expect.any(Array) }
             );
         });
     });
+
+    it("should'nt open the show page caused by click propagation", async () => {
+        render(<WithAList />);
+        const startedUrl = global.window.location.pathname;
+        const resetButton = await screen.findByRole('button', {
+            name: 'Reset views',
+        });
+        screen.getByText('500');
+        fireEvent.click(resetButton);
+        expect(screen.queryByText('500')).not.toBeNull();
```

Comment:
I would have expected this to be null, on the contrary, since, now, the counter should be 0 and no longer 500

Reply:
Yes but 500 contains 0 two times. Instead of this line I could test something like this         
`expect(screen.getAllByText('0').length).equal(1);`

Reply:
No, that's not how `getByText` works. It looks for a node in the DOM containing this exact text. Partial matches won't be returned (unless you specifically tell the RTL to do so).

You can safely use `await screen.findByText('0');` here.